packages/core-react/src/hooks/useAutoConnect.ts

```diff
@@ -0,0 +1,102 @@
+import { useClient, useAccount, Chain } from 'wagmi';
+import { connect, disconnect } from 'wagmi/actions';
+import type { ConnectResult, Connector } from '@wagmi/core';
+import { useContext, useEffect, useRef } from 'react';
+import { WalletAdapterData } from '@reef-knot/types';
+import { AcceptTermsModalContext } from '../context/acceptTermsModal';
+import { LS_KEY_TERMS_ACCEPTANCE } from '../constants/localStorage';
+import { getUnsupportedChainError } from '../helpers/getUnsupportedChainError';
+
+const isTermsAccepted = () => {
+  if (typeof window !== 'undefined') {
+    return window.localStorage?.getItem(LS_KEY_TERMS_ACCEPTANCE) === 'true';
+  }
+  return false;
+};
+
+export const useAutoConnect = (
+  autoConnectEnabled: boolean,
+  walletDataList: WalletAdapterData[],
+  chains: Chain[],
+) => {
+  const isAutoConnectCalled = useRef(false);
+  const client = useClient();
+  const { isConnected } = useAccount();
+  const { acceptTermsModal } = useContext(AcceptTermsModalContext);
+  const autoConnectOnlyAdapters = walletDataList.filter(
+    (adapter) => adapter.autoConnectOnly,
+  );
+
+  useEffect(() => {
+    if (isConnected || !autoConnectEnabled || isAutoConnectCalled.current)
+      return;
+
+    void (async () => {
```

Comment:
What do you think about breaking the logic in useEffect into separate functions? Now it is quite difficult to understand what is happening in this hook due to the large number of conditions

Reply:
Will do